### PR TITLE
Added validation for deeplabv3+

### PR DESCRIPTION
**Pull request by soumik12345**
Hi @fchollet , as per your suggestion mentioned in the previous PR (https://github.com/keras-team/keras-io/pull/619#issuecomment-915434874), I made the following changes:

1. Extended the training subset to 1000 images.
2. Added a validation set of 50 images.
3. Added a `validation_data` dataset in the call to `fit()`.
4. Added inference for both train and validation images.

Please find the corresponding Colab notebook [here](https://colab.research.google.com/drive/1D_chASKoLqfneEzWQYGRbFLN-yvOyjiG?usp=sharing).

### CODE REVIEW

File: examples/vision/deeplabv3_plus.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
 """
-Title: Multiclass semantic segmentation using DeepLabV3+
-Author: [Soumik Rakshit](http://github.com/soumik12345)
-Date created: 2021/08/31
-Last modified: 2021/09/1
-Description: Implement DeepLabV3+ architecture for Multi-class Semantic Segmentation.
+Title: FILLME
```
Comment (fchollet) [inline examples/vision/deeplabv3_plus.py:2]: Please avoid introducing any changes other than the code and image changes needed!

File: examples/vision/deeplabv3_plus.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+Author: FILLME
+Date created: FILLME
+Last modified: FILLME
+Description: FILLME
 """
 """
-## Introduction
+ Introduction
```
Comment (fchollet) [inline examples/vision/deeplabv3_plus.py:9]: E.g. formatting changes, etc.

**Review by fchollet — COMMENTED** [review]
Awesome, thank you. The code changes look good! The PR itself will need some cleanup. Please also include the new images in the PR (and the updated md and ipynb files).

File: examples/vision/deeplabv3_plus.py (hunk 1 above)
Comment (soumik12345) [reply examples/vision/deeplabv3_plus.py]: Removed unnecessary formatting changes introduced in notebook conversion.

File: examples/vision/deeplabv3_plus.py (hunk 2 above)
Comment (soumik12345) [reply examples/vision/deeplabv3_plus.py]: Removed unnecessary formatting changes introduced in notebook conversion.

**Review by fchollet — COMMENTED** [review]
Looks good, thank you! Please update the md and ipynb files.

**Comment by soumik12345** [comment]
Hi @fchollet , updated the generated files.

**Review by fchollet — APPROVED** [review]
Excellent -- we should be ready to publish. Thank you again for the great contribution.
